core-plugins/src/main/java/co/cask/hydrator/plugin/realtime/sink/RealtimeKinesisStreamSink.java

```diff
@@ -0,0 +1,177 @@
+
+/**
+ * A {@link RealtimeSink} that writes data to a Amazon Kinesis stream.
+ * If Kinesis Stream does not exist, it will be created using properties provided with this sink.
+ */
+@Plugin(type = RealtimeSink.PLUGIN_TYPE)
+@Name("KinesisSink")
+@Description("Real-time sink that outputs to a specified AWS Kinesis stream.")
+public class RealtimeKinesisStreamSink extends RealtimeSink<StructuredRecord> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RealtimeSink.class);
+  private final KinesisConfig config;
+  private static AmazonKinesisClient kinesisClient;
+
+  public RealtimeKinesisStreamSink(KinesisConfig config) {
+    this.config = config;
+  }
+
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.name),
+                                "Stream name should be non-null, non-empty.");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.awsAccessKey),
+                                "Access Key required");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.awsAccessSecret),
+                                "Access Key secret required");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.partition),
+                                "partition name should be non-null, non-empty.");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.bodyField),
+                                "Please map the input field containing data");
+    BasicAWSCredentials awsCred = new BasicAWSCredentials(config.awsAccessKey, config.awsAccessSecret);
+    kinesisClient = new AmazonKinesisClient(awsCred);
+    createAndWaitForStreamToBecomeAvailable(kinesisClient,
+                                            config.name,
+                                            config.shardCount);
+  }
+
+  @Override
+  public int write(Iterable<StructuredRecord> structuredRecords, DataWriter dataWriter) throws Exception {
+    int numRecordsWritten = 0;
+    for (StructuredRecord structuredRecord : structuredRecords) {
+      Schema schema = structuredRecord.getSchema();
+      Object data = structuredRecord.get(config.bodyField);
+      if (data == null) {
+        LOG.debug("Found null data. Skipping record.");
+        continue;
+      }
+      Schema.Field dataSchemaField = schema.getField(config.bodyField);
+      PutRecordRequest putRecordRequest = new PutRecordRequest();
+      putRecordRequest.setStreamName(config.name);
+      putRecordRequest.setPartitionKey(config.partition);
+
+      switch (dataSchemaField.getSchema().getType()) {
+        case BYTES:
+          numRecordsWritten += writeBytes(putRecordRequest, data);
+          break;
+        case STRING:
+          numRecordsWritten += writeString(putRecordRequest, data);
+          break;
+        default:
+          LOG.debug("Type {} is not supported for writing to stream", data.getClass().getName());
```

Comment:
Kinesis only supports bytes and strings? Does that mean before each Kinesis sync, you need to use a projection plugin to convert any other types? Should you add the conversion for the primitive types to string here?